### ISSUE TITLE
No information displayed in Signs.

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#235]
Every sign in game shows empty message.
![Безымянный11](https://user-images.githubusercontent.com/55348946/69231167-4a200f80-0b91-11ea-9541-c1e219057a32.jpg)

**Comment by thdankert** [comment]
Probably in `src/fheroes2/heroes/heroes_action.cpp` in `ActionToPickupResource`?

**Comment by ihhub** [comment]
Hi @thdankert , it's in `void ActionToSign(Heroes & hero, u32 obj, s32 dst_index)` function.

**Comment by thdankert** [comment]
Thanks, I would like to have a look at it then, if nobody else is working on it.

**Comment by thdankert** [comment]
I started debugging this:
- `tile.GetObjectUID` is not implemented for OBJ_SIGN
- `world.GetMapObject` therefore returns NULL, leading to the empty dialog

Digging deeper, I noticed that the `world.map_objects` does contain a sign object - but only one, regardless of the number of signs in the map.
Looking into the MP2 loading, each sign gets the same index `Point(-1, -1)` and the same uid `0` - so each sign overwrites the existing old one in `map_objects`.

I'm not familiar enough with the engine, but this does not seem right.
